lib/upipe-modules/upipe_rtp_feedback.c

```diff
@@ -0,0 +1,1034 @@
+        return UBASE_ERR_ALLOC;
+    struct upipe_rtpfb *upipe_rtpfb = upipe_rtpfb_from_upipe(upipe);
+    uref_free(upipe_rtpfb->flow_def_input);
+    upipe_rtpfb->flow_def_input = flow_def_dup;
+    upipe_rtpfb_store_flow_def(upipe, flow_def);
+
+    flow_def_dup = uref_dup(flow_def);
+    if (unlikely(flow_def_dup == NULL)) {
+        upipe_throw_fatal(upipe, UBASE_ERR_ALLOC);
+        return UBASE_ERR_ALLOC;
+    }
+    if (upipe_rtpfb->rtpfb_output)
+        upipe_rtpfb_output_store_flow_def(upipe_rtpfb->rtpfb_output, flow_def_dup);
+
+    return UBASE_ERR_NONE;
+}
+
+static int upipe_rtpfb_set_option(struct upipe *upipe, const char *k, const char *v)
+{
+    struct upipe_rtpfb *upipe_rtpfb = upipe_rtpfb_from_upipe(upipe);
+    struct uprobe *uprobe = upipe_rtpfb->uprobe;
+
+    if (!k || !v) {
+        return UBASE_ERR_INVALID;
+    } else if (!strcmp(k, "latency")) {
+        upipe_rtpfb->latency = atoi(v);
+        upipe_dbg_va(upipe, "Setting latency to %"PRIu64" msecs",
+                upipe_rtpfb->latency);
+    } else
+        return UBASE_ERR_INVALID;
+
+    return UBASE_ERR_NONE;
+}
+
+/** @internal @This processes control commands on a rtpfb pipe.
+ *
+ * @param upipe description structure of the pipe
+ * @param command type of command to process
+ * @param args arguments of the command
+ * @return an error code
+ */
+static int _upipe_rtpfb_control(struct upipe *upipe, int command, va_list args)
+{
+    switch (command) {
+        case UPIPE_REGISTER_REQUEST: {
+            struct urequest *request = va_arg(args, struct urequest *);
+            return upipe_rtpfb_alloc_output_proxy(upipe, request);
+        }
+        case UPIPE_UNREGISTER_REQUEST: {
+            struct urequest *request = va_arg(args, struct urequest *);
+            return upipe_rtpfb_free_output_proxy(upipe, request);
+        }
+        case UPIPE_GET_FLOW_DEF: {
+            struct uref **p = va_arg(args, struct uref **);
+            return upipe_rtpfb_get_flow_def(upipe, p);
+        }
+        case UPIPE_SET_FLOW_DEF: {
+            struct uref *flow_def = va_arg(args, struct uref *);
+            return upipe_rtpfb_set_flow_def(upipe, flow_def);
+        }
+        case UPIPE_GET_SUB_MGR: {
+            struct upipe_mgr **p = va_arg(args, struct upipe_mgr **);
+            return upipe_rtpfb_get_sub_mgr(upipe, p);
+        }
+        case UPIPE_GET_OUTPUT: {
+            struct upipe **p = va_arg(args, struct upipe **);
+            return upipe_rtpfb_get_output(upipe, p);
+        }
+        case UPIPE_SET_OUTPUT: {
+            struct upipe *output = va_arg(args, struct upipe *);
+            return upipe_rtpfb_set_output(upipe, output);
+        }
+        case UPIPE_SET_OPTION: {
+            const char *k = va_arg(args, const char *);
+            const char *v = va_arg(args, const char *);
+            return upipe_rtpfb_set_option(upipe, k, v);
+        }
+        case UPIPE_CONTROL_LOCAL:
```

Comment:
Generally the idea is to have a pipe-specific control, and an inline primitive in the header file.